packages/react-provider/src/transports/base.ts

```diff
@@ -1,37 +1,53 @@
 import { ICourierEvent, Interceptor } from "./types";
 
+enum ListenerType {
```

Comment:
I really, really hate typescript enums. 

`type ListinerType = "message" | "event";`

Is my preferred method